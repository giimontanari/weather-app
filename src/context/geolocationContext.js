/* Bookstores */
import React, { useState, useEffect, useRef } from "react"
import PropTypes from "prop-types"
import axios from "axios"

const API_KEY = "e9d0d195a4336ef6d447af5549072a6f"

export const GeolocationContext = React.createContext()

const GeolocationProvider = props => {
    const [geolocation, setGeolocation] = useState()
    const [geoWeather, setGeoWeather] = useState()
    const prevGeo = useRef();

    const getWeather = async () => {
        let response = await axios(`http://api.openweathermap.org/data/2.5/onecall?lat=${geolocation.lat}&lon=${geolocation.lon}&lang=es&exclude=hourly,minutely&appid=${API_KEY}`)
        setGeoWeather(response.data)
    }

    const getLocationAndWeather = async () => {
        let response = await axios("http://ip-api.com/json/?fields=status,message,country,countryCode,region,regionName,city,lat,lon,timezone,isp")
        setGeolocation(response.data);
    }
    
    useEffect(() => {
        getLocationAndWeather();
    }, [])

    useEffect(() => {
        if (prevGeo.current !== geolocation) {
            prevGeo.current = geolocation
            getWeather();
        }
    }, [geolocation])

  return <GeolocationContext.Provider value={{ geolocation, geoWeather, getWeather, setGeolocation }}>{props.children}</GeolocationContext.Provider>
}

export default GeolocationProvider

GeolocationProvider.propTypes = {
  children: PropTypes.node.isRequired
}
