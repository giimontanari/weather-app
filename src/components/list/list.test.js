import React from "react"
import renderer from 'react-test-renderer';
import List from "./list"
import { GeolocationContext } from "../../context/geolocationContext"

describe("Check list structure with object required", () => {
    
  const jsGeoWeather = {
    daily: [
      {
        dt: 31231231213,
        temp: {
          day: 213,
          min: 322,
          max: 432
        },
        humidity: 19
      }
    ]
  }

  it('renders correctly', () => {
    
      const tree = renderer
        .create(<GeolocationContext.Provider value={{ geoWeather: jsGeoWeather }}><List /></GeolocationContext.Provider>)
          .toJSON();
        
        expect(tree).toMatchSnapshot();
    });
})


