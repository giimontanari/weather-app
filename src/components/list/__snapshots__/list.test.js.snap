// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check list structure with object required renders correctly 1`] = `
<div
  className="sc-gsTCUz duZxYP"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div
        className="MuiPaper-root MuiCard-root sc-bdfBwQ kgpwnT MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="card-container"
        >
          <h1
            className="card-container-title"
          >
            miércoles, 05 septiembre, 2959
          </h1>
          <h3
            className="card-container-temp"
          >
            -60
            °
          </h3>
        </div>
        <div
          className="card-others"
        >
          <span
            className="card-others-detail"
          >
            Min: 
            49
            °
          </span>
          <hr
            className="MuiDivider-root card-others-divider MuiDivider-middle MuiDivider-light"
          />
          <span
            className="card-others-detail"
          >
            Máx: 
            159
            °
          </span>
          <hr
            className="MuiDivider-root card-others-divider MuiDivider-middle MuiDivider-light"
          />
          <span
            className="card-others-detail"
          >
            Hum: 
            19
            %
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
